lib/OpenQA/WebAPI/Controller/API/V1/User.pm

```diff
@@ -0,0 +1,46 @@
+package OpenQA::WebAPI::Controller::API::V1::User;
+use Mojo::Base 'Mojolicious::Controller';
+
+=pod
+
+=head1 NAME
+
+OpenQA::WebAPI::Controller::API::V1::User
+
+=head1 SYNOPSIS
+
+  use OpenQA::WebAPI::Controller::API::V1::User;
+
+=head1 DESCRIPTION
+
+Implements User API.
+
+=head1 METHODS
+
+=over 4
+
+=item delete()
+
+Deletes an existing user.
+
+=back
+
+=cut
+
+sub delete {
+    my ($self) = @_;
+
+    my $users = $self->schema->resultset('Users');
+    my $user  = $users->find($self->param('id'));
+
+    return $self->render(json => {error => 'Not found'}, status => 404) unless $user;
+
+    my $nickname = $user->{nickname};
+    my $ret      = $user->delete();
+    return $self->render(json => {error => 'Some error occurred'}, status => 500) unless $ret;
```

Comment:
I wonder what error we're expecting here... wouldn't we rather get exceptions if the database can't write or dies? Which of course result in a 500 by default. So for those we shouldn't need that.

Reply:
if deleting user failed, why not show the error message? we should not show the database error in UI directly?

Reply:
I'm not sure that we have an error message. I'm just saying "some error" is not better than the default error.

Reply:
And [delete](https://metacpan.org/pod/DBIx::Class::ResultSet#delete) does not seem to return 0 to indicate errors.

Reply:
After checking that. without this line the error code is 500. And I agree that 'some error ocurred' is not adding more information that the DB could give.